File: python/popen.py
#!/usr/bin/python3

import subprocess
import sys
#print(sys.argv[1])
#name=sys.argv[1]
#print(name)
pipe = subprocess.Popen("ls -aFl "+sys.argv[1], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
#(sout, serr) = pipe.communicate()
rea = pipe.stdout.readline()
co=dco=fco=0
num = []
dir = []
ch = []
while rea:
	rea = pipe.stdout.readline()
#	print(rea)
	con=0
	read = str(rea)
	
	if(read == "b''"): break
	
	result = read.split(" ")
	count = len(result)
	if(co>=2):
		if (result[0][2:3] == "d"): dco += 1
		elif (result[0][2:3] == "-"): fco += 1
	ch.append(result[0][2:])
	for i in range(0, count):
		if(result[i] != ""):
			con += 1
			if(con==5): 
				num.append(result[i])
			elif(con==9):
				dir.append(result[i][0:-3])
	co += 1

print ("directory: "+str(dco)+", "+"file: "+str(fco))	   
for j in range(2, co):
	print(dir[j]+" " + num[j]+" " + ch[j])

File: php/popen.php
#!/usr/bin/php

<?
#@extract($_GET);
#$para = $_REQUEST['para'];
#$name = $_REQUEST['name'];
#$_SERVER['argv'][1] = $_GET['name'];
$handle = popen("ls -aFl /root", "r");
#$read = fread($handle, 2096);
#echo $read;
$co = 0;
$dco = 0;
$fco = 0;
if($handle){
	while(($buffer = fgets($handle, 4096)) !== false){
	#	echo "$buffer";
		$result=explode(" ", $buffer);
		$con = 0;
		$count = count($result);
		$ch[$co] = $result[0];
		if($co >= 3){	
			if(substr($ch[$co], 0, 1) == "d") $dco++;
			else if(substr($ch[$co], 0, 1) == "-") $fco++;
		}
		for($i=1; $i<$count; $i++){
			if($result[$i] <> NULL){
				$con = $con + 1;
				if($con == 4){
					$nu[$co] = $result[$i];
					#echo "$nu\n";
				}
				else if($con == 8){
					$rm = substr($result[$i], 0, -1);
					$dir[$co] = $rm;
					#echo "$dir\n";
				}
			}
		}
		$co++;
		#echo "$dir\n";	
		#echo "$result[0]\n";
	}
	echo "directory: $dco, file: $fco\n";
	for($j=3; $j<$co; $j++){
		echo "$dir[$j] $nu[$j] $ch[$j]\n";
	}

}

pclose($handle);
?>
